src/compiler/Stage/Optimize.elm

```diff
@@ -12,34 +13,41 @@ optimize project =
     Boilerplate.optimizeProject optimizeExpr project
 
 
-optimizeExpr : Typed.Expr -> Typed.Expr
-optimizeExpr expr =
+optimizeExpr : Typed.LocatedExpr -> Typed.LocatedExpr
+optimizeExpr =
     Typed.transformAll
         [ optimizePlus
         , optimizeIfLiteralBool
         ]
-        expr
 
 
-optimizePlus : Typed.Expr -> Maybe Typed.Expr
-optimizePlus ( expr, _ ) =
-    case expr of
-        Typed.Plus ( Typed.Literal (Literal.Int left), _ ) ( Typed.Literal (Literal.Int right), _ ) ->
-            Just
-                ( Typed.Literal (Literal.Int (left + right))
-                , Type.Int
-                )
+optimizePlus : Typed.LocatedExpr -> Maybe Typed.LocatedExpr
+optimizePlus located =
+    case Typed.getExpr located of
+        Typed.Plus l r ->
+            case ( Typed.getExpr l, Typed.getExpr r ) of
+                ( Typed.Literal (Literal.Int left), Typed.Literal (Literal.Int right) ) ->
+                    Just
+                        (Located.replaceWith
+                            ( Typed.Literal (Literal.Int (left + right))
+                            , Type.Int
+                            )
+                            located
```

Comment:
It's a bit sad that something previously so simple now is so bloated with the location info, but yeah I don't know how to make it better. :thinking: